Sources/Sentry/SentryProfiler.mm

```diff
@@ -231,28 +236,25 @@
     auto metrics = serializedMetrics;
 
 #    if SENTRY_HAS_UIKIT
-    const auto framesTracker = SentryDependencyContainer.sharedInstance.framesTracker;
```

Comment:
In this hunk, we now use a private copy of the data in  `SentryDependencyContainer.sharedInstance.framesTracker.currentFrames` that is copied/stored per profiler instance. That way, when the profiler running for the last in-flight transaction is stopped, we can reset the `SentryFramesTracker`'s version of the data, while keeping the copies in instances of profilers that may still be waiting in memory for their transactions to finish.